gdc_ng_models/models/batch.py

```diff
@@ -0,0 +1,131 @@
+import json
+import sqlalchemy
+from sqlalchemy import orm
+from sqlalchemy.ext import declarative
+
+Base = declarative.declarative_base()
+
+
+class Batch(Base):
+    """Batch class to represent a collection of nodes.
+
+    Attributes:
+        id: unique identifier for the batch
+        name: name given to the batch
+        description: optional description for the batch
+        project_id: project the batch is a part of
+        created_datetime: the date and time when the batch is created
+    """
+
+    __tablename__ = "batch"
+    __mapper_args__ = {"eager_defaults": True}
+
+    # start sequence at high number to avoid collisions with existing batch_id
+    id_seq = sqlalchemy.Sequence("batch_id_seq", metadata=Base.metadata, start=1000)
+
+    id = sqlalchemy.Column(
+        sqlalchemy.BigInteger,
+        nullable=False,
+        primary_key=True,
+        server_default=id_seq.next_value(),
+    )
+    name = sqlalchemy.Column(sqlalchemy.Text, nullable=False)
+    description = sqlalchemy.Column(sqlalchemy.Text, nullable=True)
+    project_id = sqlalchemy.Column(sqlalchemy.String(64), nullable=False, index=True)
+    created_datetime = sqlalchemy.Column(
+        sqlalchemy.DateTime(timezone=True),
+        nullable=False,
+        server_default=sqlalchemy.text("now()"),
+    )
+
+    members = orm.relationship(
+        "BatchMembership",
+        back_populates="batch",
+        cascade="all, delete, delete-orphan",
+        passive_deletes=True,
+    )
+
+    def __repr__(self):
+        return "<Batch(id='{}', name='{}', description='{}', project_id='{}', created_datetime='{}')>".format(
+            self.id,
+            self.name,
+            self.description,
+            self.project_id,
+            self.created_datetime.isoformat(),
```

Comment:
Since this value is set by the server, the sqlalchemy object will have a `None` value initially, which could make this call fail.
```suggestion
            self.created_datetime.isoformat() if self.created_datetime else None,
```

Reply:
i forgot that, good catch